lib/tmdb/client.rb

```diff
@@ -0,0 +1,220 @@
+          updated_at: Time.current
+        )
+      rescue ActiveRecord::RecordInvalid => error
+        raise Error.new("#{movie.title} failed update. #{error.message}")
+      end
+
+      def common_actors_between_movies(movie_one_title, movie_two_title)
+        movie_one_results = movie_search(movie_one_title)
+        movie_two_results = movie_search(movie_two_title)
+        not_found_message = movie_one_results.not_found_message.presence || movie_two_results.not_found_message.presence
+
+        if not_found_message.present?
+          OpenStruct.new(not_found_message: not_found_message)
+        else
+          movie_one = Tmdb::Client.movie(movie_one_results.movies.first.tmdb_id)
+          movie_two = Tmdb::Client.movie(movie_two_results.movies.first.tmdb_id)
+          OpenStruct.new(
+            movie_one: movie_one,
+            movie_two: movie_two,
+            common_actors: movie_one.actors & movie_two.actors,
+            not_found_message: nil
+          )
+        end
+      end
+
+      def person_autocomplete(query)
+        data = get_parsed_multi_search_results(query)
+        person_results = data.select{ |result| result[:media_type] == "person"}
+        person_results.map{ |result| result[:name] }.uniq
+      end
+
+      def person_detail_search(person_id)
+        bio_results = get_parsed_person_bio(person_id)
+        movie_credits_results = get_parsed_person_movie_credits(person_id)
+        tv_credits_results = get_parsed_person_tv_credits(person_id)
+
+        OpenStruct.new(
+          person_id: person_id,
+          profile: MoviePersonProfile.parse_result(bio_results),
+          movie_credits: MoviePersonCredits.parse_result(movie_credits_results),
+          tv_credits: TVPersonCredits.parse_result(tv_credits_results)
+        )
+      end
+
+      def tv_actor_appearance_credits(credit_id)
+        data = get_parsed_credit(credit_id)
+        TVActorCredit.parse_record(data)
+      end
+
+      def tv_series_autocomplete(query)
+        data = get_parsed_tv_search_results(query)
+        data.map{ |d| d[:name] }.uniq
+      end
+
+      def tv_series_search(query)
+        data = get_parsed_tv_search_results(query)
+        TVSeries.parse_search_records(data) if data.present?
+      end
+
+      def tv_series(series_id)
+        data = get_parsed_tv_series_data(series_id)
+        TVSeries.parse_record(data, series_id)
+      end
+
+      def tv_season(series:, season_number:)
+        season_data = get_parsed_tv_season_data(series: series, season_number: season_number)
+        TVSeason.parse_record(
+          series: series,
+          season_data: season_data
+        )
+      end
+
+      def tv_episode(series_id:, season_number:, episode_number:)
+        episode_data = get_parsed_tv_episode_data(
+          series_id: series_id,
+          season_number: season_number,
+          episode_number: episode_number
+        )
+        TVEpisode.parse_record(episode_data)
+      end
+
+      private
```

Comment:
These private methods are the closest to the API bone -- they hold the knowledge of the url string and do the JSON parsing. These methods are reused in the public methods above. Moving this logic into private methods made it easier to test the public methods because I could mock them.